py-polars/polars/lazyframe/engine_config.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+
+    from rmm.mr import DeviceMemoryResource  # type: ignore[import-not-found]
+
+
+class GPUEngine:
```

Comment:
Any reason not to use a dataclass here?

Reply:
Not otherwise used in polars, so not wanting to introduce another import dependency to keep import short (which AFAIU is a stated goal)

Reply:
It is. :)